lib/inline_oneboxer.rb

```diff
@@ -74,6 +73,12 @@ def self.lookup(url, opts = nil)
 
   private
 
+  def self.domain_is_blocked?(hostname)
+    SiteSetting.blocked_onebox_domains&.split('|').any? do |blocked|
+      hostname.match("(^|\\.)#{Regexp.quote(blocked)}$")
```

Comment:
mixed on this ... but we could save on the regex with:

`"a.b.c".end_with?(".b.c") || "a.b.c" == ".b.c"` 

Do upper case domains get a bypass here? Can they be injected somehow?

Reply:
oh also if not going without regex ... be sure to use `match?` which returns a bool and is faster. 

```
> "a".match? /a/
=> true
```

Was something I pushed nobu to add :)

Reply:
I'll have a look and see which is neater